Problem: eth signer not accurate in state machine

yihuang opened the pull request:
Solution:
- cleanup the eth signer related codes
- state machine always use `ethtypes.MakeSigner` based on chain config and block number, it matters when we do block number based hardfork and the signer logic is changed in that, it don't matter in `x/upgrade` style upgrade though.
- optimize rpc api to reuse the `From` field in msg, instead of always recover the sender.

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

yihuang commented on CHANGELOG.md:
```diff
@@ -65,6 +65,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [#454](https://github.com/crypto-org-chain/ethermint/pull/454) Migrate transient stores to object stores.
 * (evm) [#461](https://github.com/crypto-org-chain/ethermint/pull/461) Remove custom signer getter, use the option in protobuf file instead.
 * (evm) [#469](https://github.com/crypto-org-chain/ethermint/pull/469) More aggressive cache of common parameters in object store.
+* (evm) [#]() State machine always use `ethtypes.MakeSigner` based on current chain config and block number, this is
```
```suggestion
* (evm) [#470](https://github.com/crypto-org-chain/ethermint/pull/470) State machine always use `ethtypes.MakeSigner` based on current chain config and block number, this is
```

mmsqe commented on rpc/backend/utils.go:
```diff
@@ -104,7 +104,7 @@ func (b *Backend) getAccountNonce(accAddr common.Address, pending bool, height i
 			}
 
-			sender, err := ethMsg.GetSenderLegacy(b.chainID)
+			sender, err := ethMsg.GetSenderLegacy(ethtypes.LatestSignerForChainID(b.chainID))
```
should we use MakeSigner as well

yihuang replied:
it's fine in json-rpc I think.

mmsqe commented on rpc/types/utils.go:
```diff
@@ -167,27 +167,30 @@ func NewTransactionFromMsg(
 		signer = ethtypes.LatestSignerForChainID(tx.ChainId())
 	} else {
-		signer = ethtypes.HomesteadSigner{}
+		signer = ethtypes.FrontierSigner{}
```
oh does this check wrong from beginning?

yihuang replied:
i think so

yihuang replied:
we are still not consistency in handling the legacy transactions across the json-rpc, anyway 🤦

Commit:
Problem: eth signer not accurate in state machine

Solution:
- cleanup the eth signer related codes
- state machine always use `ethtypes.MakeSigner` based on chain config
  and block number, it matters when we do block number based hardfork
  and the signer logic is changed in that.
- optimize rpc api to reuse the `From` field in msg, instead of always
  recover the sender.

Update CHANGELOG.md

Signed-off-by: yihuang <huang@crypto.com>

fix tests

fix test
